dot/types/grandpa.go

```diff
@@ -196,3 +196,62 @@ type GrandpaVote struct {
 func (v GrandpaVote) String() string {
 	return fmt.Sprintf("hash=%s number=%d", v.Hash, v.Number)
 }
+
+// GrandpaEquivocation is used to create a proof of equivocation
+type GrandpaEquivocation struct {
+	RoundNumber     uint64
+	ID              [32]byte
+	FirstVote       GrandpaVote
+	FirstSignature  [64]byte
+	SecondVote      GrandpaVote
+	SecondSignature [64]byte
+}
+
+// GrandpaEquivocationEnum is a wrapper object for GRANDPA equivocation proofs, useful for unifying prevote
+// / and precommit equivocations under a common type.
+type GrandpaEquivocationEnum scale.VaryingDataType
+
+// Set sets a VaryingDataTypeValue using the underlying VaryingDataType
+func (ge *GrandpaEquivocationEnum) Set(value scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*ge)
+	err = vdt.Set(value)
+	if err != nil {
+		return err
+	}
+	*ge = GrandpaEquivocationEnum(vdt)
+	return nil
+}
+
+// Value will return the value from the underlying VaryingDataType
+func (ge *GrandpaEquivocationEnum) Value() (value scale.VaryingDataTypeValue, err error) {
+	vdt := scale.VaryingDataType(*ge)
+	return vdt.Value()
+}
+
+// NewGrandpaEquivocation returns a new VaryingDataType to represent a grandpa Equivocation
+func NewGrandpaEquivocation() *GrandpaEquivocationEnum {
+	vdt := scale.MustNewVaryingDataType(PreVoteEquivocation{}, PreCommitEquivocation{})
+	ge := GrandpaEquivocationEnum(vdt)
+	return &ge
+}
+
+// PreVoteEquivocation equivocation type for a prevote
+type PreVoteEquivocation GrandpaEquivocation
```

Comment:
```suggestion
type Prevote GrandpaEquivocation
```
the type infers that it is an equivocation